packages/react-three-cannon/src/debug-provider.tsx

```diff
@@ -36,9 +37,18 @@ export const DebugProvider: FC<DebugProviderProps> = ({
 
   useFrame(() => {
     for (const uuid in bodyMap) {
-      refs[uuid].matrix.decompose(v, q, s)
-      bodyMap[uuid].position.copy(v as unknown as Vec3)
-      bodyMap[uuid].quaternion.copy(q as unknown as CQuaternion)
+      const ref = refs[uuid]
+      if (ref instanceof InstancedMesh) {
+        const index = parseInt(uuid.split('/')[1])
+        ref.getMatrixAt(index, m)
+        m.decompose(v, q, s)
+        bodyMap[uuid].position.copy(v as unknown as Vec3)
+        bodyMap[uuid].quaternion.copy(q as unknown as CQuaternion)
+      } else {
+        ref.matrix.decompose(v, q, s)
+        bodyMap[uuid].position.copy(v as unknown as Vec3)
+        bodyMap[uuid].quaternion.copy(q as unknown as CQuaternion)
+      }
```

Comment:
Sorry, last one:

```suggestion
      const matrix = ref instanceof instancedMesh
        ? ref.getMatrixAt(parseInt(uuid.split('/')[1]))
        : ref.matrix
      matrix.decompose(v, q, s)
      bodyMap[uuid].position.copy(v as unknown as Vec3)
      bodyMap[uuid].quaternion.copy(q as unknown as CQuaternion)
```

It's good to keep the branches to minimum.

Reply:
Oof, I'm sorry, I messed that up. I'll cherry pick your previous commit.